File: src/main/java/ru/arc/hooks/PartiesHook.java
package ru.arc.hooks;

import com.alessiodp.parties.api.Parties;
import com.alessiodp.parties.api.interfaces.PartiesAPI;
import com.alessiodp.parties.api.interfaces.Party;
import com.alessiodp.parties.api.interfaces.PartyPlayer;

import java.util.UUID;

public class PartiesHook {

    PartiesAPI api;

    public PartiesHook(){
        api = Parties.getApi();
    }


    public String description(UUID playerUuid){
        PartyPlayer player = api.getPartyPlayer(playerUuid);
        if(player == null || !player.isInParty()) return null;
        Party party = api.getParty(player.getPartyId());
        return party.getDescription();
    }

    public String name(UUID playerUuid){
        PartyPlayer player = api.getPartyPlayer(playerUuid);
        if(player == null || !player.isInParty()) return null;
        Party party = api.getParty(player.getPartyId());
        return party.getName();
    }

    public String tag(UUID playerUuid){
        PartyPlayer player = api.getPartyPlayer(playerUuid);
        if(player == null || !player.isInParty()) return null;
        UUID uuid = player.getPartyId();
        if(uuid == null) return null;
        Party party = api.getParty(player.getPartyId());
        if(party == null) return null;
        return party.getTag();
    }

    public String color(UUID playerUuid){
        PartyPlayer player = api.getPartyPlayer(playerUuid);
        if(player == null || !player.isInParty()) return "";
        if (player.getPartyId() == null) return "";
        Party party = api.getParty(player.getPartyId());
        if(party == null || party.getColor() == null) return "";
        String code = party.getColor().getCode();
        return code == null ? "" : code;
    }


}

File: src/main/kotlin/ru/arc/hooks/PartiesHook.kt
package ru.arc.hooks

import com.alessiodp.parties.api.Parties
import org.bukkit.Bukkit
import java.util.UUID

class PartiesHook {

    private val api = Parties.getApi()

    fun description(playerUuid: UUID): String? {
        val player = api.getPartyPlayer(playerUuid) ?: return null
        if (!player.isInParty) return null
        val partyId = player.partyId ?: return null
        return api.getParty(partyId)?.description
    }

    fun name(playerUuid: UUID): String? {
        val player = api.getPartyPlayer(playerUuid) ?: return null
        if (!player.isInParty) return null
        val partyId = player.partyId ?: return null
        return api.getParty(partyId)?.name
    }

    fun tag(playerUuid: UUID): String? {
        val player = api.getPartyPlayer(playerUuid) ?: return null
        if (!player.isInParty) return null
        val partyId = player.partyId ?: return null
        return api.getParty(partyId)?.tag
    }

    fun color(playerUuid: UUID): String {
        val player = api.getPartyPlayer(playerUuid) ?: return ""
        if (!player.isInParty) return ""
        val partyId = player.partyId ?: return ""
        val party = api.getParty(partyId) ?: return ""
        return party.color?.code ?: ""
    }

    /** Null means a party member other than the leader attempted a network launch. */
    fun localLaunchParticipants(playerUuid: UUID): Set<UUID>? {
        val player = api.getPartyPlayer(playerUuid) ?: return setOf(playerUuid)
        if (!player.isInParty) return setOf(playerUuid)
        val partyId = player.partyId ?: return setOf(playerUuid)
        val party = api.getParty(partyId) ?: return setOf(playerUuid)
        if (party.leader != playerUuid) return null
        return party.members.mapNotNullTo(linkedSetOf()) { memberId ->
            Bukkit.getPlayer(memberId)?.takeIf { it.isOnline }?.uniqueId
        }.also { it += playerUuid }
    }
}
